src/hardware/vga.cpp

```diff
@@ -42,40 +42,44 @@ uint32_t ExpandTable[256];
 uint32_t Expand16Table[4][16];
 uint32_t FillTable[16];
 
-std::pair<std::string, std::string> VGA_DescribeType(const VGAModes type, uint16_t mode)
+std::tuple<std::string, uint16_t, std::string> VGA_DescribeMode()
```

Comment:
Firstly, now the name of function is not accurate (it should be `VGA_DescribeCurrentMode`), secondly, I very much prefer using pure functions by default unless there's a compelling reason not to.

In this case, converting this pure function that takes a mode and returns its description in human-readable format into a non-pure function is unfortunate. This limits the function's usefulness (now you can only use it to describe the current mode, and nothing else).

Pure functions are also preferable due to testability (previously, you could simply test the function in isolation by passing various values into it and inspecting the return value; now testing would require the whole global context to be created and manipulated).

Arguably, the original function could be improved so it only takes a single `mode` parameter.

Reply:
Happy to switch it back to a pure function 👍

(The reason I internalized the variables, vga.mode and CurMode, are was because they're globals to the VGA routines, which have unfortunately been "spilled" into SDL main. 

But that's a separate issue (and quite out of scope from this little PR), and hopefully will be mopped up with the re-write anyway.

Reply:
Yeah and in this case it might not make a huge difference; I was mostly talking about general principles 😄 In general, I like to default to better approaches to "set an example", so to speak.

Reply:
This version should hit the mark! 
This (should) be the last of the backports to 0.80.1 (unless any new issues bubble up).